src/utils/filterContests.js

```diff
@@ -3,14 +3,18 @@ import type { Contest } from '../types';
 
 export function filterContests(contests: Array<Contest>, from?: Date, to?: Date): Array<Contest> {
   contests = contests.filter((contest) => {
-    if (from instanceof Date && contest.startTime instanceof Date)
-      return contest.startTime >= from;
-    else return false;
+    if (from instanceof Date)
+      if (contest.startTime instanceof Date)
+        return contest.startTime >= from;
+      else return false;
+    else return true;
```

Comment:
If I understood correctly, the behaviour is: get everything if there is no `from`, otherwise exclude those starting before and those without start time.

There is a simple way to do this:
```
if (!from instanceof Date) return true;
if (!contest.startTime instanceof Date) return false;
return contest.startTime >= from
```

joke aside, that is what the guy meant by avoiding the `else` keyword

Reply:
Actually, I think it would also be better if the `from` parameter was checked before applying the filter, instead of inside the filter function. Note that these comments apply to the filter for `to` also.

Reply:
About checking `from` from outside the filter function, flow did not seem to recognize that `from` was not undefined when I did that, this solution was the only thing that worked, but I'll look into it.

Reply:
Yeah, that really seems to invalidate the type. The solution suggested by flow on these scenarios is to create a new variable inside the type checking.
```
if (from instanceof Date) {
  let localFrom: Date = from;
  contests.filter(/* something that uses localFrom */);
}
```